content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -504,6 +504,33 @@ stops adjusting the target (and sets the `ScalingActive` Condition on itself
 to `false`) until you reactivate it by manually adjusting the target's desired
 replica count or HPA's minimum replica count.
 
+### Migrating Deployments and StatefulSets to horizontal autoscaling
+
+When an HPA is enabled, it is recommended that the value of `spec.replicas` of
+the Deployment and/or Statefulset object is removed.  Without doing so, anytime
+a change to that object is applied, example via `kubectl apply -f
+deployment.yaml`, will instruct Kubernetes to scale the current number of Pods
+to the desired value retained in the `spec.replicas` key. This may not be
+desired and could be troublesome when an HPA is active.
+
+Keep in mind that the removal of `spec.replicas` may incur a one time
+degradation of Pod counts as the default value of this key is 1 (reference
+[Deployment Replicas](/docs/concepts/workloads/controllers/deployment#replicas).
+Upon the update all Pods except 1 will begin their Termination procedures.  Any
+deployment application afterwards will behave as normal and respect a Rolling
+Update configuration as desired.  We can avoid this degradation using two
+options:
```

Comment:
These aren't freely interchangeable options--which to use depends on whether the user is currently using client-side or server-side apply to manage the object in question.